### ISSUE TITLE
Missing leading slash: no error, but wrong filename

### ISSUE THREAD

**Issue by ddfreyne** [nanoc/nanoc#1015]
A routing rule that returns a path that does not start with a slash will cause the filename to be stupidly concatenated to the output dir name.

See [this blog post](https://wob.site/articles/nanoc-part-1/):

> One major gotcha is that the beginning slash is necessary for globs and paths. If you don’t include a beginning slash for globs and paths, Nanoc will end up concatenating the name of the default base directory it’s meant to use (like content or output) with the glob or path (yeah, I find this dumb – just remember the beginning slash).

For example, the following:

```ruby
compile '/foo.*' do
  filter :erb, stuff: self
  write 'foo.html'
end
```

… writes `outputfoo.html`.

**Comment by ddfreyne** [comment]
CC @caseywilliams — I found this bug on your “Using Nanoc 4” article.

**Comment by ddfreyne** [comment]
Fixed in #1016.

**Comment by caseywilliams** [comment]
@ddfreyne, wow, you're amazing! Thank you for this and all of your excellent work on Nanoc! I've updated my article, too :)
